pkg/operator/ceph/object/status.go

```diff
@@ -76,3 +79,10 @@ func updateStatusBucket(client client.Client, name types.NamespacedName, phase c
 
 	logger.Debugf("object store %q status updated to %v", name, phase)
 }
+
+func buildStatusInfo(cephObjectStore *cephv1.CephObjectStore) map[string]string {
+	m := make(map[string]string)
+	m["endpoint"] = buildDNSEndpoint(BuildDomainName(cephObjectStore.Name, cephObjectStore.Namespace), cephObjectStore.Spec.Gateway.Port)
```

Comment:
> Port cannot be empty, it is always mandatory and SecurePort is optional.

Wait, the non-secure port is optional last I knew. I also see checks like [this](https://github.com/rook/rook/blob/eabedd7f79dd6602ab1f011693096d54cf39c697/pkg/operator/ceph/object/spec.go#L192-L196) where the secure endpoint is generated. Seems like we need to add both the http and https endpoints to the status, depending on whether the `port`, `securePort`, or both are set.

Reply:
Hum I was also looking at the API validation https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/common.yaml#L429-L432, which makes me think the `Port` is mandatory.

Reply:
hmm that min should be set to 0 since the code allows it to be optional. This shows that the securePort must not be used very often or else we should have heard about the bug by now.

Reply:
I remember this is something we've had issues to address correctly in the past. It was one of @egafford's PR.